include/lsst/afw/typehandling/SimpleGenericMap.h

```diff
@@ -57,37 +55,45 @@ class SimpleGenericMap final : public MutableGenericMap<K> {
 public:
     SimpleGenericMap() = default;
     SimpleGenericMap(SimpleGenericMap const& other) = default;
-    SimpleGenericMap(SimpleGenericMap&&) = default;
-    SimpleGenericMap(GenericMap<K> const& other) : _storage(_convertStorage(other)) {}
+    SimpleGenericMap(SimpleGenericMap&&) noexcept = default;
+    SimpleGenericMap(GenericMap<K> const& other) : _storage(_convertStorage(other)), _keyView(other.keys()) {}
     virtual ~SimpleGenericMap() noexcept = default;
 
-    SimpleGenericMap& operator=(SimpleGenericMap const& other) = default;
-    SimpleGenericMap& operator=(SimpleGenericMap&&) = default;
+    SimpleGenericMap& operator=(SimpleGenericMap const& other) {
+        std::vector<K> newKeys = other._keyView;
+        _storage = other._storage;
+        // no exceptions past this point
```

Comment:
It makes it easier to verify the code's exception-safety (and reminds anybody editing the code to pay attention to exceptions). In this case:

- if the copy-construction of `newKeys` fails, then the map is unchanged ✔️ 
- if the copy-assignment of `_storage` fails, then `_storage`, and therefore the map, should be unchanged (though now I can't seem to find an exception-safety spec for `unordered_map` 😟, maybe I should also copy-and-swap it)
- since none of the later operations can throw, if the assignment of `_storage` succeeds, the map assignment as a whole will succeed ✔️ 

Would it help if I said something like "strong exception safety because no exceptions past this point"?

Reply:
Yes, that's more helpful. It was not entirely clear whether the existing comment was supposed to be "there can be none" or "please be careful", or something else.

Or, maybe: "strong exception safety because no exceptions can occur past this point" ?